style/train/classifier_trainer.py

```diff
@@ -1,28 +1,135 @@
-from typing import List
+from pathlib import PosixPath
 
+from sklearn import metrics
+from sklearn.decomposition import TruncatedSVD
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.linear_model import LogisticRegression
+from sklearn.linear_model import SGDClassifier
+from sklearn.model_selection import GridSearchCV
+from sklearn.model_selection import train_test_split
+from sklearn.naive_bayes import MultinomialNB
 from sklearn.pipeline import Pipeline
+from sklearn.svm import LinearSVC
 
-from style.predict.servable.base import BaseServable
+from style.constants import FILE_PATH_BOOK_DB
+from style.constants import MODEL_EXPORT_PATH
+from style.dataset.reader import Dataset
+from style.dataset.reader import DatasetReader
+from style.predict.servable.base import SklearnBasedClassifierServable
+
+
+class TextNormalizer:
+    pass
+
+
+def split_dataset(
+    dataset: Dataset,
+    test_percentage: float = 0.2,
+    random_state: int = 42,
+):
+    docs_train, docs_test, y_train, y_test = train_test_split(
+        dataset.data,
+        dataset.target,
+        test_size=test_percentage,
+        random_state=random_state,
+    )
+    return docs_train, docs_test, y_train, y_test, dataset.target
 
 
 def train_sklearn_classification_model(
-    data: List[str],
-    export_path: str,
+    docs_train,
+    docs_test,
+    y_train,
+    y_test,
+    dataset_target,
     pipeline: Pipeline,
-    params: dict,
-    train_percentage: float = 0.8,
-    test_percentage: float = 0.2,
-) -> BaseServable:
+    grid_search_params: dict,
+):
     """This method trains multiple models via GridSearchCV and it creates a servable
     from the best model.
 
+    See https://scikit-learn.org/stable/tutorial/statistical_inference/putting_together.html for more details.
+
+
+    """
+
+    grid_search = GridSearchCV(pipeline, grid_search_params, n_jobs=-1, cv=2)
```

Comment:
`cv` can be parametric. can you do that? the default can be `5` but let's send `2` in the `run` for now.